app/assets/javascripts/user.js

```diff
@@ -223,4 +223,62 @@ $(document).ready(function () {
   $("#read_tou").on("click", function () {
     $("#continue").prop("disabled", !($(this).prop("checked") && $("#read_ct").prop("checked")));
   });
+
+
+  const isTaken /* Record<string, string> */ = {}
+
+  function indicateInvalidDisplayName(isValid /* boolean | undefined*/) {
+    // isValid might also be 'undefined', hence we check explicitly for 'true' and 'false'
+    $("#user_display_name").toggleClass("is-valid", isValid === true)
+    $("#user_display_name").toggleClass("is-invalid", isValid === false)
+  }
+
+  function checkDisplayName(){
+    const displayName = $("#user_display_name").val()
+    if(displayName === ""){
```

Comment:
Might check instead if the name is shorter than 3 characters.

Reply:
We don't really want to be duplicating validations on the client as they will just get out of sync.

This is why it would probably better to have a specific backend method in the rails code that could be called with a proposed username and which would apply all the validations the user model will eventually apply.

Reply:
Right now if I start typing the name, this code will say that the name is valid after the first typed character. But it's obviously invalid.

Reply:
> We don't really want to be duplicating validations on the client as they will just get out of sync.
> This is why it would probably better to have a specific backend method in the rails code that could be called with a proposed username and which would apply all the validations the user model will eventually apply.

Totally agree with this, but I have no idea how to do this. Even better would be to give a (translated) reason why it is not valid.

However, this is already a step in the right direction...

Reply:
Last change of minimal display name length was [in 2006](https://github.com/openstreetmap/openstreetmap-website/commit/37bd2971b49da9364fe5dd25415e97e38b34abd7#diff-9802ca3c9c4cf89904fd44bc114e35ebdf2c5dd3d5b645491e2b253e1afef29bR10). Getting out of sync here is very unlikely. If you think that it is, you can just avoid validating short names by not doing requests and not adding the `is-valid` class.